smarts/core/local_agent.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2020. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+import logging
+import time
+from concurrent import futures
+from typing import Tuple
+from smarts.core.buffer_agent import BufferAgent
+from smarts.core.agent import Agent
+
+from smarts.zoo.agent_spec import AgentSpec
+
+
+class LocalAgent(BufferAgent):
+    """A remotely controlled agent."""
+
+    def __init__(self):
+        # Track the last action future.
+        #self._act_future = None
+        self._agent = None
+        self._agent_spec = None
+
+    def act(self, obs):
+        """Call the agent's act function asynchronously and return a Future."""
+        act_future = futures.Future()
```

Comment:
Theoretically, we should be using an `Executor` to create this `Future` but it is still less of a problem to do it this way even though the documentation argues against it.